pkg/collector/corechecks/oracle-dbm/oracle.go

```diff
@@ -0,0 +1,214 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package oracle
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/oracle-dbm/common"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/oracle-dbm/config"
+	"github.com/DataDog/datadog-agent/pkg/obfuscate"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/version"
+	_ "github.com/godror/godror"
+	"github.com/jmoiron/sqlx"
+	go_ora "github.com/sijms/go-ora/v2"
+)
+
+// Check represents one Oracle instance check.
+type StatementsFilter struct {
+	SQLIDs map[string]int
+	//ForceMatchingSignatures map[uint64]int
+	ForceMatchingSignatures map[string]int
+}
+
+type Check struct {
+	core.CheckBase
+	config                                  *config.CheckConfig
+	db                                      *sqlx.DB
+	hostname                                string
+	dbmEnabled                              bool
+	agentVersion                            string
+	checkInterval                           float64
+	tags                                    []string
+	cdbName                                 string
+	statementsFilter                        StatementsFilter
+	statementMetricsMonotonicCountsPrevious map[StatementMetricsKeyDB]StatementMetricsMonotonicCountDB
+	dbHostname                              string
+	dbVersion                               string
+	driver                                  string
+}
+
+// Run executes the check.
+func (c *Check) Run() error {
+	if c.db == nil {
+		db, err := c.Connect()
+		if err != nil {
+			c.Teardown()
+			return err
+		}
+		c.db = db
+	}
+
+	if c.hostname == "" {
+		hostname, err := os.Hostname()
+		if err != nil {
+			return fmt.Errorf("failed to get hostname: %w", err)
+		}
+		c.hostname = hostname
```

Comment:
Is this bit necessary? Checks usually do not send hostname explicitly, deferring to the default hostname in the sender. The agent has many ways to infer hostname, which often doesn't match `os.Hostname`, that would make hostname on these metrics different from other checks.

Reply:
I removed the code that retrieves the hostname and I'm passing the empty string instead.